### PR TITLE
Update umbra-js encryption

### LINKED ISSUE #41: Specify size of random number to generate
To compute a stealth address for the recipient, the sender must generate a random number. Currently a 32-byte random number is used.

As part of the spec, determine if we should stick with a 32-byte random number or use a different size.

### PR DESCRIPTION
**Pull request by mds1**
Closes #41, #30, #31

Changes:
- We now generate 16 byte random numbers, stored in a 32 byte value. The first 16 bytes can be optionally used as a payload extension, as described in #41 and #30
- Encryption is now done using XOR as described in #31
- Switch from npm to yarn and setup prettier

### CODE REVIEW

File: umbra-js/classes/RandomNumber.js [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
    */
-  constructor(length = 32) {
-    this.length = length;
-    this.value = utils.shuffled(utils.randomBytes(length));
-  }
+  constructor(payloadExtension = zeroPrefix) {
+    // Generate default random number without payload extension
+    this.randomLength = 16; // 16 byte random number
+    this.fullLength = 32; // Pad random number to 32 bytes
+    const randomNumberAsBytes = utils.shuffled(utils.randomBytes(this.randomLength));
```
Comment (apbendi) [inline umbra-js/classes/RandomNumber.js:21]: Not a change, just wondering, what does the `shuffled` function do here?

**Review by apbendi — APPROVED** [review]
Looks great, tests pass. Nice work! One small question that is unrelated to the change, but I'm just curious. Good to merge :)

File: umbra-js/classes/RandomNumber.js (hunk 1 above)
Comment (mds1) [reply umbra-js/classes/RandomNumber.js:21]: It just shuffles the random bytes array with a Fisher–Yates shuffle. I figured I'd add that in to get a bit more randomness, but in reality it probably doesn't make much difference

Comment (apbendi) [reply umbra-js/classes/RandomNumber.js:21]: Gotcha! I'd say that doesn't hurt, but doesn't really help either. Either the bytes coming out of `randomBytes` are secure randomness as is, or they're not. No need to change, though. Should be good to merge 👍

**Pull request merged**